<launch>
<!-- Arguments Start -->

  <!-- Simulation -->
  <arg name="runsim"                default="true"/>
  <arg name="log"                   default="true"/>
  
  <!-- Mission -->
  <arg name="mission_start_x"       default="-7"/>
  <arg name="mission_start_y"       default="1.5"/>
  <arg name="mission_goal_x"        default="7"/>
  <arg name="mission_goal_y"        default="1.5"/>

  <!-- Environment -->
  <arg name="world_x_min"           default="-10"/>
  <arg name="world_y_min"           default="-10"/>
  <arg name="world_z_min"           default="0.2"/>
  <arg name="world_x_max"           default="10"/>
  <arg name="world_y_max"           default="10"/>
  <arg name="world_z_max"           default="0.9"/>
  <arg name="world_resolution"      default="0.1"/>
  
  <!-- InitTrajPlanner Parameters -->
  <arg name="ecbs_w"                default="1.3"/>
  <arg name="grid_xy_res"           default="0.5"/>
  <arg name="grid_z_res"            default="1.0"/>
  <arg name="grid_margin"           default="0.1"/>

  <!-- Corridor Parameters-->
  <arg name="box_xy_res"            default="0.1"/>
  <arg name="box_z_res"             default="0.1"/>

  <!-- RBPPlanner Parameters -->
  <arg name="plan_time_step"        default="1"/>

<!-- Arguments End -->

<!-- Nodes Start -->

  <node pkg = "mpc_atypical" type = "chomp_test_app" name = "chomp_test" output = "screen">
    <param name="log"                        value="$(arg log)" />

    <param name="mission/start_x"            value="$(arg mission_start_x)" />
    <param name="mission/start_y"            value="$(arg mission_start_y)" />
    <param name="mission/goal_x"             value="$(arg mission_goal_x)" />
    <param name="mission/goal_y"             value="$(arg mission_goal_y)" />

    <param name="world/x_min"                value="$(arg world_x_min)" />
    <param name="world/y_min"                value="$(arg world_y_min)" />
    <param name="world/z_min"                value="$(arg world_z_min)" />
    <param name="world/x_max"                value="$(arg world_x_max)" />
    <param name="world/y_max"                value="$(arg world_y_max)" />
    <param name="world/z_max"                value="$(arg world_z_max)" />

    <param name="ecbs/w"                     value="$(arg ecbs_w)" />
    <param name="grid/xy_res"                value="$(arg grid_xy_res)" />
    <param name="grid/z_res"                 value="$(arg grid_z_res)" />
    <param name="grid/margin"                value="$(arg grid_margin)" />

    <param name="box/xy_res"                 value="$(arg box_xy_res)" />
    <param name="box/z_res"                  value="$(arg box_z_res)" />

    <param name="plan/time_step"             value="$(arg plan_time_step)" />
    <rosparam command="load" file="$(find mpc_atypical)/params/chomp_param_map3.yaml" />
  
  </node>

    

	<!--visualization module-->
	<!-- octomap server -->
	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server" output="screen" args="$(find mpc_atypical)/worlds/map_reduced_tmp3.bt"> 

		<param name="resolution" value="0.1" />
		<param name="frame_id" type="string" value="world" />
		<param name="sensor_model/max_range" value="4.0" />
		<param name="save_directory" value="$(optenv OCTOMAP_SAVE_DIR ./)" />
		<param name="track_changes" value="true"/>
		<param name="listen_changes" value="false"/>
		<param name="topic_changes" value="/octomap_tracking_server/changeset" />
		<!-- added by JBS -->
		
		<param name="height_map" value="false" />
		
		<param name="color/r" value="0.4" />
		<param name="color/g" value="0.4" />
		<param name="color/b" value="0.4" />
		<param name="color/a" value="0.3" />
		
		<param name="occupancy_max_z" value="" />
		<param name="occupancy_min_z" value="0.2" />
		<param name="filter_ground" value="true" />
		<param name="base_frame_id" value="/world" />
		
		<param name="min_change_pub" value="0" />
	</node>
	
	
	 <!--visualization module-->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mpc_atypical)/rviz/rviz.rviz"/>  
	
<!-- Nodes End -->	
</launch>